accio-sqlrewrite/src/main/java/io/accio/sqlrewrite/ModelSqlRender.java

```diff
@@ -147,35 +116,115 @@ private void processRelationshipInfos(Column column, Model baseModel, List<Expre
                     .collect(toSet()));
 
             // Add select items based on the type of column
-            selectItems.add(getSelectItemsExpression(column, alias));
+            if (calculatedFieldRelationshipInfo.isAggregated()) {
+                selectItems.add(getSelectItemsExpression(column, Optional.of(calculatedFieldRelationshipInfo.getAlias())));
+            }
+            else {
+                selectItems.add(getSelectItemsExpression(column, Optional.of(getRelationableAlias(baseModel.getName()))));
+            }
+        }
+        else {
+            // No relationships, add select item with no alias
+            selectItems.add(getSelectItemsExpression(column, Optional.empty()));
+            columnWithoutRelationships.put(column.getName(), column.getExpression().get());
         }
     }
 
     @Override
-    protected String getCalculatedSubQuery(Model baseModel, CalculatedFieldRelationshipInfo relationshipInfo)
+    protected List<SubQueryJoinInfo> getCalculatedSubQuery(Model baseModel, List<CalculatedFieldRelationshipInfo> relationshipInfos)
     {
-        String requiredExpressions = format("%s AS \"%s\"",
-                RelationshipRewriter.rewrite(relationshipInfo.getExpressionRelationshipInfo(), parseExpression(relationshipInfo.getColumn().getExpression().get())),
-                relationshipInfo.getAlias());
+        ImmutableList.Builder<SubQueryJoinInfo> queries = ImmutableList.builder();
+        queries.addAll(getToOneRelationshipsQuery(baseModel, relationshipInfos));
+        queries.addAll(getToManyRelationshipsQuery(baseModel, relationshipInfos));
```

Comment:
Consider rename `getToOneRelationshipsQuery` and `getToManyRelationshipsQuery`.
We distinguish them according to doing aggregation or not. Maybe rename them to
- `getMapExpressionQuery`: allow only to_one relationship
- `getReduceExpressionQuery`: allow both to_one and to_many
Inspire form the concept of `map` and `reduce`.

Reply:
I still prefer the original method name, will add more comment to methods.